linter/lints/common.go

```diff
@@ -35,3 +37,50 @@ func GetExtWithOID(exts []pkix.Extension, oid asn1.ObjectIdentifier) *pkix.Exten
 	}
 	return nil
 }
+
+// ReadOptionalASN1BooleanWithTag attempts to read and advance incoming to
+// search for an optional DER-encoded ASN.1 element tagged with the given tag.
+// Unless outExpectedTagPresent or outPresent are nil, it stores whether the tag
+// was found in outExpectedTagPresent and an element with the tag was found in
+// outPresent, otherwise each boolean will take the default value. It reports
+// whether all reads were successful.
+func ReadOptionalASN1BooleanWithTag(incoming *cryptobyte.String, outExpectedTagPresent *bool, outPresent *bool, tag cryptobyte_asn1.Tag, defaultValue bool) bool {
+	if outExpectedTagPresent != nil {
+		*outExpectedTagPresent = defaultValue
+	}
+	if outPresent != nil {
+		*outPresent = defaultValue
+	}
+	if incoming.Empty() {
+		return false
+	}
```

Comment:
I think this extra check is not necessary. `ReadOptionalASN1` will return false if incoming is empty, right?

Reply:
It does not! https://go.dev/play/p/hrdH7WjjnoR

Reply:
Oh right, this makes sense! We talked about it live, but the conclusion was:

ReadOptionalASN1 will always return true if the incoming buffer is empty, because that just means "this optional field was not present; and that's okay."

The conclusion was still that we don't need this check, because we actually want to follow the same behavior: if the incoming buffer is empty, we should return success, and set `*out` to `defaultValue`.